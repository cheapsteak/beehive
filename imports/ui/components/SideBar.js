// Framework
import React from 'react';

// Components
import { Container, Row, Col, ButtonGroup } from 'reactstrap';

import { GameButton, QuitButton } from './SideBar.style';
/**
 * A list of field rows with a header.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const SideBar = ({ numberOfBees, createBee, removeAllBees }) => {
  const createBeeType = type => ({ type });
  const createWorker = () => createBee(createBeeType('Worker'));
  const createDrone = () => createBee(createBeeType('Drone'));
  const createQueen = () => createBee(createBeeType('Queen'));
  const removeBees = () => removeAllBees();

  return (
    <aside>
      <Container className="side-bar" fluid>
        <Row>
          <Col>
            <ButtonGroup size="sm">
              <GameButton onClick={createWorker}>Worker</GameButton>
              <GameButton onClick={createDrone}>Drone</GameButton>
              <GameButton onClick={createQueen}>Queen</GameButton>
              <QuitButton onClick={removeBees}>X</QuitButton>
            </ButtonGroup>
          </Col>
        </Row>
      </Container>
    </aside>
  );
};

export default SideBar;
