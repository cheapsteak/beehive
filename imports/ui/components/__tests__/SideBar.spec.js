import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SideBar from '../SideBar';
import { GameButton, QuitButton } from '../SideBar.style';

describe('<SideBar />', () => {
  const defaultProps = {
    numberOfBees: 4,
    createBee: jest.fn(),
    removeAllBees: jest.fn()
  };

  describe('SideBar SnapShot', () => {
    it('renders SideBar correctly', () => {
      const tree = ReactTestRenderer.create(
        <SideBar {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('SideBar Props', () => {
    let wrapper, buttons, xbutton;

    beforeEach(() => {
      const wrapper = shallow(<SideBar {...defaultProps} />);
      buttons = wrapper.find(GameButton);
      xbutton = wrapper.find(QuitButton).first();
    });

    it('should call createBee', () => {
      buttons.forEach((node, index) => {
        node.simulate('click');
      });
      expect(defaultProps.createBee).toHaveBeenCalledTimes(3);
    });

    it('should call removeAllBees', () => {
      xbutton.simulate('click');
      expect(defaultProps.removeAllBees).toHaveBeenCalled();
    });
  });
});
