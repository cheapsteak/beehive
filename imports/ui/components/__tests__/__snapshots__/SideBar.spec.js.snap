// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> SideBar SnapShot renders SideBar correctly 1`] = `
<aside>
  <div
    className="side-bar container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="btn-group-sm btn-group"
          role="group"
        >
          <button
            className="sc-bdVaJa btn-block bDWFJH btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            Worker
          </button>
          <button
            className="sc-bdVaJa btn-block bDWFJH btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            Drone
          </button>
          <button
            className="sc-bdVaJa btn-block bDWFJH btn btn-secondary"
            onClick={[Function]}
            type="button"
          >
            Queen
          </button>
          <button
            className="sc-bwzfXH btn-block uTxCW btn btn-danger"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</aside>
`;
